src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -116,73 +134,119 @@ public ActivityTraceFlags TraceFlags
+        /// enabled or <c>{OriginalFormat}</c> attribute (message template) is
+        /// not found.
+        /// </remarks>
         public string? FormattedMessage
         {
-            get => this.Data.Message;
-            set => this.Data.Message = value;
+            get => this.ILoggerData.FormattedMessage;
+            set => this.ILoggerData.FormattedMessage = value;
         }
 
         /// <summary>
-        /// Gets or sets the raw state attached to the log. Set to <see
-        /// langword="null"/> when <see
+        /// Gets or sets the log body.
+        /// </summary>
+        /// <remarks>
+        /// Note: Set to the <c>{OriginalFormat}</c> attribute (message
+        /// template) if found otherwise the formatted log message.
+        /// </remarks>
+        public string? Body
+        {
+            get => this.Data.Body;
+            set => this.Data.Body = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the raw state attached to the log.
+        /// </summary>
+        /// <remarks>
+        /// Note: Set to <see langword="null"/> when <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled.
+        /// </remarks>
+        [Obsolete("State cannot be accessed safely outside of an ILogger.Log call stack. Use Attributes instead to safely access the data attached to a LogRecord. State will be removed in a future version.")]
+        public object? State
+        {
+            get => this.ILoggerData.State;
+            set => this.ILoggerData.State = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the state values attached to the log.
         /// </summary>
-        public object? State { get; set; }
+        /// <remarks><inheritdoc cref="Attributes" /></remarks>
+        [Obsolete("Use Attributes instead StateValues will be removed in a future version.")]
+        public IReadOnlyList<KeyValuePair<string, object?>>? StateValues
+        {
+            get => this.Attributes;
+            set => this.Attributes = value;
+        }
 
         /// <summary>
-        /// Gets or sets the parsed state values attached to the log. Set when <see
+        /// Gets or sets the attributes attached to the log.
+        /// </summary>
+        /// <remarks>
+        /// Note: Set when <see
+        /// cref="OpenTelemetryLoggerOptions.IncludeAttributes"/> is enabled and
+        /// log record state implements <see cref="IReadOnlyList{T}"/> or <see
+        /// cref="IEnumerable{T}"/> of <see cref="KeyValuePair{TKey, TValue}"/>s
+        /// (where TKey is <c>string</c> and TValue is <c>object</c>) or <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled
         /// otherwise <see langword="null"/>.
-        /// </summary>
-        public IReadOnlyList<KeyValuePair<string, object?>>? StateValues { get; set; }
+        /// </remarks>
+        public IReadOnlyList<KeyValuePair<string, object?>>? Attributes { get; set; }
```

Comment:
Just to make `LogRecord` closer to spec. The way the upcoming "bridge API" thing works is you can make a bridge/appender that is able to submit `LogRecord` directly through the provider (including attributes). The goal here is to set it up so both sides can speak "Attributes" which is nice and consistent with the spec.

Reply:
This will make it more complex for people authoring Redaction/similar logic. Instead of state, statevalues, now they need to look into Attributes too. 
Even if we call it StateValues, it wont block the bridge API implementation, right?

Reply:
Clarified offline that this is backward compatible, State/StateValues are aliasing to Attributes only.